cupy/_core/core.pyx

```diff
@@ -2544,20 +2564,33 @@ cdef _ndarray_base _array_default(obj, dtype, order, Py_ssize_t ndmin):
         return a
     cdef Py_ssize_t nbytes = a.nbytes
 
+    cdef pinned_memory.PinnedMemoryPointer mem
     stream = stream_module.get_current_stream()
-    # Note: even if obj is already backed by pinned memory, we still need to
-    # allocate an extra buffer and copy from it to avoid potential data race,
-    # see the discussion here:
-    # https://github.com/cupy/cupy/pull/5155#discussion_r621808782
-    cdef pinned_memory.PinnedMemoryPointer mem = (
-        _alloc_async_transfer_buffer(nbytes))
-    if mem is not None:
-        src_cpu = numpy.frombuffer(mem, a_dtype, a_cpu.size)
-        src_cpu[:] = a_cpu.ravel(order)
-        a.data.copy_from_host_async(mem.ptr, nbytes)
-        pinned_memory._add_to_watch_list(stream.record(), mem)
+
+    cdef intptr_t ptr_h = <intptr_t>(a_cpu.ctypes.data)
+    if pinned_memory.is_memory_pinned(ptr_h):
+        if blocking:
+            warnings.warn(
+                'pinned memory is in use but blocking is set to True, an '
+                'unnecessary synchronization would happen',
```

Comment:
I suppose the same argument (that blocking could be necessary) is also applicable here. Need to phrase it better.

Reply:
Removed the warning in commit 6b962f7283e4870811e0024163f42063cfe32f30